### PR TITLE
Added the implementation of Hash Tables

### LINKED ISSUE #170: I would like add implementation of Hash table using C
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Implementation of hash tables 

**Languages**
-   C

### PR DESCRIPTION
**Pull request by Sashi445**
# Title/Heading
A simple implementation of hash tables

## Type of change

Fix : #170 

# Checklist:
-    [x] I have performed a self-review of my own code
-    [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: C/HashTableImplementation.c [hunk 1]
```diff
@@ -0,0 +1,112 @@
+
```
Comment (kaustubhgupta) [inline C/HashTableImplementation.c]: Remove extra spaces from code

File: C/HashTableImplementation.c [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+int main(){
+	int base = 10; // Took default base as 10 it can be any value
+	node* hashTable[base]; //Initializing hash table
+	hashTableNull(hashTable, base); //making it null before insertion
+	for(int i=0; i<50; i++){
+		insert(rand()%100+1, hashTable, base);
+	}
+	printHashTable(hashTable, base);	
+}
```
Comment (kaustubhgupta) [inline C/HashTableImplementation.c:96]: Update readme of this language folder

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Review by YashSaxena9 — APPROVED** [review]
Looks fine if we talk about logical implementation, can be improved by optimizing rehashing and use of templates :)

**Comment by YashSaxena9** [comment]
@Sashi445, please make the changes as suggested by @kaustubhgupta

File: C/HashTableImplementation.c (hunk 1 above)
Comment (Sashi445) [reply C/HashTableImplementation.c]: Okay!!

File: C/HashTableImplementation.c (hunk 2 above)
Comment (Sashi445) [reply C/HashTableImplementation.c:96]: Yeah I will

File: C/README.md [hunk 3]
```diff
@@ -13,4 +13,4 @@ C is a general-purpose, high-level language that was originally developed by Den
 |    Stack implementation    |                 [link](./StackImplementation.c)                  |
 |  Sum of digits of number   |                  [link](./sumOfNumberDigits.C)                   |
 |  Binary Search Tree        |                  [link](./BinarySearchTree.C)                    |
-|  Floyd's Cycle Detection   |                  [link](./floyds_cycle_detection_for_linked_lists.c)				|
+|  Hash Table Implementation |                  [link](./HashTableImplementation.c)             |
```
Comment (kaustubhgupta) [inline C/README.md:17]: By mistake, you removed another contributor entry! Please add it back and your PR is ready to merge!

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: C/README.md (hunk 3 above)
Comment (Sashi445) [reply C/README.md:17]: Okay I am on it

**Comment by kaustubhgupta** [comment]
Thanks for your contribution!

**Comment by Sashi445** [comment]
> Thanks for your contribution!

Thanks man you helped me make my first pull request thanks a lot!!

**Pull request merged**